src/encoded/static/components/experiment_series.js

```diff
@@ -192,21 +384,82 @@ class ExperimentSeriesComponent extends React.Component {
-                    <CartAddAllElements elements={viewableDatasets} />
+                    <CartAddAllElements elements={Object.values(viewableDatasets)} />
                 </div>
             );
 
             // Collect unique internal_tags from all relevant experiments.
-            internalTags = _.uniq(viewableDatasets.reduce((allInternalTags, experiment) => (
+            internalTags = _.uniq(Object.values(viewableDatasets).reduce((allInternalTags, experiment) => (
                 experiment.internal_tags && experiment.internal_tags.length > 0 ? allInternalTags.concat(experiment.internal_tags) : allInternalTags
             ), []));
+
+            // Experiment table data
+            Object.values(viewableDatasets).forEach((experiment) => {
+                // Select one analysis based on file assembly and file count
+                // for extracting quality metrics
+                const filesByDesiredAssembly = experiment.files.reduce(
+                    (fileAtIds, f) => {
+                        if (selectedAssembly.includes(f.assembly)) {
+                            fileAtIds.push(f['@id']);
+                        }
+                        return fileAtIds;
+                    },
+                    []
+                );
+                const analyses = experiment.analyses || [];
+                const selectedAnalysis = analyses.filter(
+                    analysis => analysis.files.every(f => filesByDesiredAssembly.includes(f))
+                ).sort((a, b) => a.length - b.length)[0] || { files: [] };
+                experiment.files = experiment.files.filter(
+                    f => selectedAnalysis.files.includes(f['@id'])
+                );
+                // Number of subrows to be shown corresponds to the number of biological replicates. Show at least one by default.
+                let bioRepNums = _.uniq(
+                    experiment.files.map(f => f.biological_replicates).flat()
+                );
+                if (bioRepNums.length === 0) {
+                    bioRepNums = [''];
+                }
+                // Extract experiment metadata
+                const rowCount = bioRepNums.length;
+                const datasetRows = [];
+                bioRepNums.sort((a, b) => a - b).forEach((num, i) => {
+                    const cellMeta = { bioRepNum: num, isAuthorized, rowCount };
+                    datasetRows.push(
+                        <tr key={experiment['@id'].concat('_', i)}>
+                            {Object.entries(columns).map(([columnId, expCol]) => (columns[columnId].replicateSpecific || i === 0 ?
+                                (expCol.display ? expCol.display(experiment, cellMeta) : <td key={columnId} rowSpan={columns[columnId].replicateSpecific ? 1 : rowCount}>{expCol.getValue ? expCol.getValue(experiment, cellMeta) : experiment[columnId]}</td>)
```

Comment:
Nested ?: aren’t allowed in AirBnB, but ESLint rarely detects it. Better to break it up (might have to use a temporary variable to hold one value).

Reply:
Is using if here OK?

Reply:
That’s often the best solution.